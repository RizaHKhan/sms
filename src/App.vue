<template>
  <div id="app">
    <div id="nav">
      <template v-for="(link, linkIndex) in links">
        <router-link
          v-text="link.title"
          :to="link.path"
          :key="'link' + linkIndex"
        />
        <span
          v-if="linkIndex < links.length - 1"
          v-text="' | '"
          :key="'|' + linkIndex"
        ></span>
      </template>
    </div>
    <router-view />
  </div>
</template>

<script>
export default {
  name: 'App',
  data: function () {
    return {
      links: [
        {
          title: 'Home',
          path: '/'
        },
        {
          title: 'About',
          path: '/about'
        }
      ]
    };
  }
};
</script>
