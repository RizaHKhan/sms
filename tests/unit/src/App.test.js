import App from '@/App.vue';

import testHelpers from '@@/testHelpers.js';

describe('App.vue', () => {
  let localVue;
  let store;
  let setup;

  let permissions = ['student'];

  beforeEach(() => {
    setup = testHelpers.setupTestVariables(permissions);
    localVue = setup.localVue;
    store = setup.store;
  });

  test('Renders correctly', async () => {
    const wrapper = await testHelpers.shallowMountWrapper(App, {
      localVue,
      store
    });

    expect(wrapper)
      .toMatchSnapshot();
  });
});
