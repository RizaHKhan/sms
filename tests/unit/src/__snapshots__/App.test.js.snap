// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Renders correctly 1`] = `
<div id="app">
  <div id="nav">
    <a>Home</a>
    <span> | </span>
    <a>About</a>
    <span> | </span>
    <a>Users</a>
    <!---->
  </div>
  <div class="wrapper">
    <router-view-stub name="default"></router-view-stub>
  </div>
</div>
`;
